ngraph/core/reference/include/ngraph/runtime/reference/binary_convolution.hpp

```diff
@@ -0,0 +1,227 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/runtime/reference/convolution.hpp"
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            namespace details
+            {
+                inline uint8_t extract_bit(uint8_t val, uint8_t bit)
+                {
+                    return (uint8_t)((val >> bit) & 0x0001);
```

Comment:
nit: May be the mask should be `0x01`? Since the range of values for `uint8_t` is `[0..255]`

Reply:
Done